src/diffpy/utils/parsers/loadmetafile.py

```diff
@@ -0,0 +1,212 @@
+
+    # json
+    if extension == '.json':
+        # dump if non-existing
+        if not existing:
+            with open(dbname, 'w') as jsonfile:
+                pdfs = entry  # for return
+                json.dump(pdfs, jsonfile, indent=2)
+
+        # update if existing
+        else:
+            with open(dbname, 'r') as json_read:
+                pdfs = json.load(json_read)
+                pdfs.update(entry)
+            with open(dbname, 'w') as json_write:
+                # dump to string first for formatting
+                json.dump(pdfs, json_write, indent=2)
+
+    return pdfs
+
+
+def markup_PDF(hddata: dict, rv: list, muname=None):
+    # FIXME: may be better suited for REST API package, not diffpy.utils
+    """Put PDF file information into a dictionary.
+
+    hddata  -- Dictionary of metadata.
+    rv      -- List of (r, gr) pairs.
+    muname  -- file to save into (default None, no saving occurs).
+
+    Returns the dictionary loaded from/into markup file.
+    """
+
+    # gather data
+    data = {}
+    data.update({'r': list(rv[:, 0]), 'gr': list(rv[:, 1])})
+    data.update(hddata)
+
+    # return directly
+    if muname is None:
+        return data
+
+    # save to disk when enabled
+    extension = pathlib.Path(muname).suffix
+    if extension not in supported_formats:
+        raise Exception(f"Format of {muname} is not supported.")
+
+    # dumps into file, automatically overwrites
+    if extension == '.json':
+        with open(muname, 'w') as json_write:
+            json.dump(data, json_write, indent=2)
+
+    return data
+
+
+def markup_oneline(filename):
+    """Reformat lists in markup languages to take up only one line.
+
+    Works well when only lists are surrounded by square brackets and no other data is comma and newline separated.
+
+    filename    -- name of markup file to reformat.
+    """
+
+    # check file type
+    extension = pathlib.Path(filename).suffix
+    if extension not in supported_formats:
+        raise Exception(f"Format of {filename} is not supported.")
+
+    if extension == '.json':
+        # cannot easily do regex substitution since lists are of floats
+        with open(filename, 'r+') as json_file:
+            lines = json_file.readlines()
+            json_file.seek(0)
+            json_file.truncate()
+
+            s_flag = False
+            for line in lines:
+                if "\"r\": [" in line or "\"gr\": [" in line:
+                    s_flag = True
+                    updated_line = line[:-1]
+                elif "]," in line:
+                    s_flag = False
+                    updated_line = f"{updated_line[:-1]}{line.strip()}\n"
+                    json_file.write(updated_line)
+                elif s_flag:
+                    updated_line += f"{line[:-1].strip()} "
+                else:
+                    json_file.write(line)
+
+
+def apply_schema_to_file(filename, schemaname, multiple_entries=False):
```

Comment:
this one may be better in the api where we will maintain the schema

Reply:
Got it, agreed